src/index.js

```diff
@@ -101,6 +102,16 @@ export default class NodeClient {
 
     maps() {
         if (this.#config.version && this.#loadMaps) return this.#maps;
-        throw new Error('Eik config was not loaded or "loadMaps" is "false" when calling .maps()');
+        throw new Error(
+            'Eik config was not loaded or "loadMaps" is "false" when calling .maps()',
+        );
+    }
+
+    mapping(dependency) {
+        let mapping = null;
+        for (const map of this.maps()) {
```

Comment:
This will continue to loop through this.maps even after found a match. You could return the result immediately - unless we can see a scenario where the same dependency exists in multiple maps and we have defined that the last import map takes presedence.

Reply:
This is a pretty interesting point when I think about it.

There might be multiple import maps and two maps (or more) can contain the same identifier but with different targets. Then the question is which one wins. In our build tool plugins the last occurrence wins at the moment (in the plugins the algorithm is a little bit different though, but the result is the same). So, should first or last occurrence win here?

The answer is probably more complex. Import Maps do have a feature to deal with this which is defined under [https://github.com/WICG/import-maps#multiple-versions-of-the-same-module](scopes). Currently our build tool plugins have a little bit of a naive implementation of the Import Map spec and we do not take scoping into account. We should have full Import Map support though (aka also supporting scoping) and we have an experimental implementation in the ESBuild plugin: https://github.com/eik-lib/esbuild-plugin/pull/71/

Ideally this method should actually apply full Import Map parsing and adhere to the algorithm in the spec to decide what to return.

But to be practical here: I think this is fine since this aligns with whats currently happening in the build tools. But, we should do an effort to get support for the whole Import Map spec and in this process get this method to do so too.

Reply:
I've left the code alone and simply updated the docs for now.